pypeit/images/combineimage.py

```diff
@@ -310,26 +269,26 @@ def run(self, bias=None, scattlight=None, flatimages=None, ignore_saturation=Fal
 
         # Build the combined image
         comb = pypeitimage.PypeItImage(image=comb_img, ivar=utils.inverse(comb_var), nimg=nframes,
-                                       amp_img=pypeitImage.amp_img, det_img=pypeitImage.det_img,
+                                       amp_img=rawImage.amp_img, det_img=rawImage.det_img,
                                        rn2img=comb_rn2, base_var=comb_basev, img_scale=comb_scl,
                                        # NOTE: This *must* be a boolean.
                                        bpm=np.logical_not(gpm), 
                                        # NOTE: The detector is needed here so
                                        # that we can get the dark current later.
-                                       detector=pypeitImage.detector,
+                                       detector=rawImage.detector,
                                        PYP_SPEC=self.spectrograph.name,
                                        units='e-' if self.par['apply_gain'] else 'ADU',
                                        exptime=comb_texp, noise_floor=self.par['noise_floor'],
                                        shot_noise=self.par['shot_noise'])
 
         # Internals
         # TODO: Do we need these?
-        comb.files = self.files
-        comb.rawheadlist = pypeitImage.rawheadlist
-        comb.process_steps = pypeitImage.process_steps
+        comb.files = file_list
+        comb.rawheadlist = rawImage.rawheadlist
+        comb.process_steps = rawImage.process_steps
 
         # Build the base level mask
-        comb.build_mask(saturation='default', mincounts='default')
+        comb.build_mask(saturation='default' if not ignore_saturation else None, mincounts='default')
```

Comment:
I don't think this is necessary given how `ignore_saturation` is used in the lines above, but I also think this is a reasonable thing just to ensure that `ignore_saturation` is adhered to if the code above changes.

Reply:
Okay I'm leaving it for now.